pkgs/pkgs-lib/formats.nix

```diff
@@ -417,4 +417,48 @@ rec {
       '';
     };
 
+    libconfig = { }: {
+      type = with lib.types;
+        let
+          valueType = nullOr (oneOf [
+            bool
+            int
+            float
+            str
+            path
+            (attrsOf valueType)
+            (listOf valueType)
+          ]) // {
+            description = "Libconfig value";
+          };
+        in valueType;
+      generate = name: value:
+        let
+          output = pkgs.writeText name (lib.generators.toLibconfig { } value);
+          validator = pkgs.runCommandCC "libconfig-validator" { buildInputs = [ pkgs.libconfig ]; } ''
+          $CC -lconfig -x c - -o "$out" <<EOF
+//    Copyright (C) 2005-2023  Mark A Lindner, ckie
```

Comment:
Neat Idea. Did you consider either `<<-EOF` or a few more lines to take the file path as an input parameter with the C in a separate file?

Reply:
I think I wanted to do something like that before it got lost in organizational hell, done now. Thanks :)

(Also, confirmed validator doesn't rebuild every time now)